### PR TITLE
Deleting a Cart

### PR DESCRIPTION
**Pull request by trc311**
Issue #18

### CODE REVIEW

**Review by WesPainter — CHANGES_REQUESTED** [review]
Modify delete to search shopping carts for proper shopping cart. Directly feeding sid won't work, as sid doesn't necessarily correlate to position in list.

**Comment by Pratik-Karnik** [comment]
@WesPainter I suggested @trc311  a few changes and helped debug his code. He should be submitting his new commit soon.

**Comment by trc311** [comment]
Yes, I'll submit both the commits in a bit.

**Comment by trc311** [comment]
Okay, modified the code. This is working. Review and Close please.

**Comment by Pratik-Karnik** [comment]
@trc311 What about the API which deletes a particular product in a particular shop cart? Where is that?

**Comment by trc311** [comment]
Yes, making a separate pull request for that.

**Review by WesPainter — CHANGES_REQUESTED** [review]
Close. This will still error out if you try and delete something that doesn't exist.  My suggestion is to move the return statement outside of the for loop and replace it with break.  This way it will return no content no matter what.

**Comment by trc311** [comment]
Yes, that works.

File: shopcart.py [hunk 1]
```diff
@@ -237,6 +237,16 @@ def create_products(sid):
...
 
 ######################################################################
+# DELETE A SHOPPING CART
+######################################################################
+@app.route('/shopcarts/<int:sid>', methods=['DELETE'])
+def delete_shopcarts(sid):
+    for i in range(len(shopping_carts)):
+    	if shopping_carts[i]['sid'] == sid:
+    	    del shopping_carts[i]
+    	    return '', HTTP_204_NO_CONTENT
```
Comment (Pratik-Karnik) [inline shopcart.py]: Do you think we should show an error message if you try to delete a cart which does not exist?

Comment (trc311) [reply shopcart.py]: It shows, HTTP_204_NO_CONTENT.
Shouldn't that be enough?

Comment (WesPainter) [reply shopcart.py]: In the slides it indicates the best practice is probably not to throw an error, just return normally.  If you try and delete it and its not there, then its deleted?

Comment (Pratik-Karnik) [reply shopcart.py]: Oh I remember him saying this in class.

Comment (trc311) [reply shopcart.py]: Yes product gets deleted, if you repeat it shows a blank screen and HTTP_204_NO_CONTENT

**Review by WesPainter — APPROVED** [review]

**Pull request merged**